### PR TITLE
Downgrade Prow to K8s `1.23` and switch to `eks-blueprints`

### PR DESCRIPTION
**Pull request by RedbackThomson**
Description of changes:
`dockershim` was removed in K8s `1.24`, and we rely on `docker` to run all of the integration tests in the cluster. For now we will need to downgrade to `1.23` to support the current set of tests. 

Our previous upgrade from `1.21` also broke the `ghproxy` deployment in the Prow Helm chart because it requires a `PersistentVolumeClaim`, which used to utilise the EKS built-in EBS CSI driver. As of `1.23`, EKS no longer provides the EBS CSI driver, meaning we now need to install this into the cluster to create PVCs backed by EBS.

Upgrading to CDK 2 also allows us to now use the `eks-blueprints` module to simplify and harden our CDK cluster configuration. It also lets us install `Karpenter` with one line of code!

The SSM parameters that we used to have in here was not being used - we removed the `DaemonSet` that we used for that a long time ago. It wasn't compatible with the EKS blueprints configuration anyway so I have stripped it here. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: infra/bin/test-ci.ts [hunk 1]
```diff
@@ -34,13 +34,13 @@ new TestCIStack(app, "TestCIStack", {
...
+    hostedZoneId:
+      app.node.tryGetContext("hosted_zone_id") || process.env.HOSTED_ZONE_ID,
   },
   logsBucketName:
     app.node.tryGetContext("logs_bucket") || process.env.LOGS_BUCKET,
   logsBucketImport:
     app.node.tryGetContext("logs_bucket_import") ||
     process.env.LOGS_BUCKET_IMPORT ||
     false,
-  pvreBucketName:
```
Comment (jljaco) [inline infra/bin/test-ci.ts:44]: Is this no longer used anywhere?

File: infra/lib/ci-cluster.ts [hunk 2]
```diff
@@ -1,58 +1,80 @@
...
+      .region(Stack.of(this).region)
+      .version(clusterVersion)
+      .clusterProvider(new blueprints.MngClusterProvider(mngProps))
+      .resourceProvider(
+        GlobalResources.HostedZone,
+        new ImportHostedZoneProvider(props.hostedZoneId)
+      )
+      .addOns(
+        new blueprints.addons.VpcCniAddOn(),
+        new blueprints.addons.KarpenterAddOn(),
```
Comment (jljaco) [inline infra/lib/ci-cluster.ts:61]: 🔥

File: infra/lib/ci-cluster.ts [hunk 3]
```diff
@@ -1,58 +1,80 @@
...
+        new blueprints.addons.EbsCsiDriverAddOn(),
+        new blueprints.addons.ExternalDnsAddOn({
+          hostedZoneResources: [GlobalResources.HostedZone],
+        })
+      )
+      .build(this, "TestInfraCluster");
+
+    this.testCluster = blueprintStack.getClusterInfo().cluster;
+
+    this.namespaceManifests = [PROW_JOB_NAMESPACE, PROW_NAMESPACE].map(
```
Comment (jljaco) [inline infra/lib/ci-cluster.ts:72]: Do we not need to create `EXTERNAL_DNS_NAMESPACE` now?  Is that handled by the `ExternalDnsAddOn`?

File: infra/package-lock.json [hunk 4]
```diff
@@ -1153,10 +1340,21 @@
...
+      },
+      "funding": {
+        "url": "https://github.com/sponsors/ljharb"
+      }
+    },
     "node_modules/aws-cdk-lib": {
-      "version": "2.63.2",
-      "resolved": "https://registry.npmjs.org/aws-cdk-lib/-/aws-cdk-lib-2.63.2.tgz",
-      "integrity": "sha512-Kk9aP79O+9UJ3Bw0wjquMD+RpWPQ/Fa4wUcP5mdpROpoipPXsFFZPTXT8j8kp2oFOZa9OHxacTF06+/M0H/bWQ==",
+      "version": "2.60.0",
```
Comment (jljaco) [inline infra/package-lock.json:1355]: Any particular reason why we need to downgrade `aws-cdk-lib` here?  Shouldn't we be using the latest?

File: infra/package.json [hunk 5]
```diff
@@ -11,19 +11,20 @@
...
-    "constructs": "^10.0.0",
-    "aws-cdk-lib": "^2.0.0"
+    "aws-cdk-lib": "2.60.0",
+    "constructs": "^10.0.0"
   },
   "devDependencies": {
     "@types/jest": "^26.0.10",
     "@types/node": "10.17.27",
-    "aws-cdk-lib": "^2.0.0",
+    "aws-cdk-lib": "^2.60.0",
```
Comment (jljaco) [inline infra/package.json:20]: IMO we should fix the version of `aws-cdk-lib` instead of using this ">=" style of dependency specification

**Review by jljaco — COMMENTED** [review]
Thank you so much for doing this -- a few minor comments

File: infra/bin/test-ci.ts (hunk 1 above)
Comment (RedbackThomson) [reply infra/bin/test-ci.ts:44]: The SSM parameters that we used to have in here was not being used - we removed the `DaemonSet` that we used for that a long time ago. It wasn't compatible with the EKS blueprints configuration anyway so I have stripped it here.

File: infra/lib/ci-cluster.ts (hunk 3 above)
Comment (RedbackThomson) [reply infra/lib/ci-cluster.ts:72]: Handled by the addon now. In fact I had it in there before and after waiting 45 minutes for the stack to create it failed on the last step - which was creating that namespace. 😠

File: infra/package-lock.json (hunk 4 above)
Comment (RedbackThomson) [reply infra/package-lock.json:1355]: Unfortunately the latest version had incompatibilities with `eks-blueprints`. Weird stuff like the `ec2.InstanceType` wouldn't work. I had to downgrade to the version they list.

File: infra/package.json (hunk 5 above)
Comment (RedbackThomson) [reply infra/package.json:20]: I agree, but we are restricted by `eks-blueprints` (see above)

**Review by jljaco — APPROVED** [review]
/lgtm (just for fun since Prow isn't listening right now ;)

File: infra/lib/ci-cluster.ts [hunk 6]
```diff
@@ -1,58 +1,80 @@
...
 } from "./test-ci-stack";
+import {
+  GlobalResources,
+  ImportHostedZoneProvider,
+} from "@aws-quickstart/eks-blueprints";
 
-export type CIClusterCompileTimeProps = ProwGitHubSecretsChartProps;
+export type CIClusterCompileTimeProps = ProwGitHubSecretsChartProps & {
+  hostedZoneId: string;
+};
```
Comment (jaypipes) [inline infra/lib/ci-cluster.ts:21]: huh.. TIL...

So the `&` operator can be used against a variable of type `object` to "append" properties to that object?

File: infra/lib/ci-cluster.ts [hunk 7]
```diff
@@ -1,58 +1,80 @@
...
+      .clusterProvider(new blueprints.MngClusterProvider(mngProps))
+      .resourceProvider(
+        GlobalResources.HostedZone,
+        new ImportHostedZoneProvider(props.hostedZoneId)
+      )
+      .addOns(
+        new blueprints.addons.VpcCniAddOn(),
+        new blueprints.addons.KarpenterAddOn(),
+        new blueprints.addons.AwsLoadBalancerControllerAddOn(),
+        new blueprints.addons.EbsCsiDriverAddOn(),
```
Comment (jaypipes) [inline infra/lib/ci-cluster.ts:63]: I think it's kinda crazy that the EBS CSI driver is no longer installed by default in EKS clusters. In fact, that is pointed out in [this article](https://medium.com/@elliotgraebert/comparing-the-top-eight-managed-kubernetes-providers-2ae39662391b) specifically as causing major headaches compared to GKE and AKS...

**Review by jaypipes — COMMENTED** [review]
@RedbackThomson thanks for doing this. I wanted to learn a bit more specifically about the changes to the DNS setup for our cluster. Can you write a couple sentences as to why those changes were needed? Thanks!

**Comment by jljaco** [comment]
This was a follow-up to https://github.com/aws-controllers-k8s/test-infra/pull/301

File: infra/lib/ci-cluster.ts (hunk 6 above)
Comment (RedbackThomson) [reply infra/lib/ci-cluster.ts:21]: Yes it's a union of the types. Typescript has some crazy type definition helpers

**Comment by RedbackThomson** [comment]
> I wanted to learn a bit more specifically about the changes to the DNS setup for our cluster. Can you write a couple sentences as to why those changes were needed?

The only differences that this PR makes to the DNS setup is that we are now installing the `external-dns` Helm chart through the `eks-blueprints` package. Part of the constructor for its add-on is a list of `HostedZoneIDs` so it can allowlist those within the controller. I guess this is to prevent the controller from overriding HostedZones it shouldn't be able to access. Otherwise, the DNS situation is the same - we have a Route53 HostedZone that is set up independently of this CDK and we have an Ingress that sets up a load balancer. The `external-dns` controller glues those together by adding records to the HostedZone with the public address of the load balancer.

File: infra/lib/ci-cluster.ts (hunk 6 above)
Comment (RedbackThomson) [reply infra/lib/ci-cluster.ts:21]: Oh `ProwGitHubSecretsChartProps` is also a `type`, though, not an `object`

**Pull request merged**